src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,84 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final int RESIZE_FACTOR = 2;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] myData;
+    private int size;
+
+    public MyHashMap() {
+        myData = new Node[INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size >= myData.length * DEFAULT_LOAD_FACTOR) {
+            resize();
+        }
+        Node<K, V> currentNode = myData[getIndex(key)];
+        if (currentNode == null) {
+            myData[getIndex(key)] = new Node<>(key, value, null);
+        }
+        while (currentNode != null) {
+            if (Objects.equals(currentNode.key, key)) {
+                currentNode.value = value;
+                return;
+            }
+            if (currentNode.next == null) {
+                currentNode.next = new Node<>(key, value, null);
+                break;
+            }
+            currentNode = currentNode.next;
+        }
+        size++;
     }
```

Comment:
Let's remove this field from redundant if() statement in the begining and move it to the bottom under the while loop. So it'll also work like a default case when `(currentNode != null)` is `false`
```suggestion
        while (currentNode != null) {
            if (Objects.equals(currentNode.key, key)) {
                currentNode.value = value;
                return;
            }
            if (currentNode.next == null) {
                currentNode.next = new Node<>(key, value, null);
                break;
            }
            currentNode = currentNode.next;
        }
        // move here
        myData[getIndex(key)] = new Node<>(key, value, null);
        size++;
    }
```